torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1600,11 +1607,14 @@ XLATensor XLATensor::rsub(const XLATensor& input, at::Scalar other,
 
 void XLATensor::s_copy_(XLATensor& input, XLATensor& src) {
   if (input.GetDevice() == src.GetDevice()) {
-    input.SetIrValue(src.GetIrValue());
+    input.SetIrValue(
```

Comment:
I would avoid adding a cast IR node if it is not needed.
So I would check for src dtype and our dtype, and avoid that.
We are also missing the other path in aten_xla_type.cpp line 2347. We need to possibly convert the tensor (only if dtype differs) before a SetTensor().

Reply:
Right, haven't noticed the other path in `AtenXlaType`.

Regarding the redundant cast, I'm pretty sure we optimize away no-op casts in the XLA back-ends. Doing manual optimizations kinda defeats the purpose of a multi-tier compiler.

Reply:
I know we can. XLA will too. But we better not have graphs polluted with noops, as they are already big enough to inspect for debug.

Reply:
Done.